crapssim/bet.py

```diff
@@ -534,6 +535,46 @@ def __repr__(self) -> str:
         return f"{self.__class__.__name__}({self.number}, amount={self.amount})"
 
 
+# Hop bets -------------------------------------------------------------------
+
+
+class Hop(Bet):
+    def __init__(self, result: tuple[int, int], amount: typing.SupportsFloat) -> None:
+        super().__init__(amount)
+        self.result: tuple[int, int] = tuple(sorted(result))
+
+    def get_result(self, table: Table) -> BetResult:
+        if table.dice.result in self.winning_results:
+            result_amount = self.payout_ratio(table) * self.amount + self.amount
+            should_remove = True
+        else:
+            result_amount = -1 * self.amount
+            should_remove = True
+        return BetResult(result_amount, should_remove)
+
+    @property
+    def is_easy(self) -> bool:
+        return self.result[0] != self.result[1]
+
+    @property
+    def winning_results(self) -> list[tuple[int, int]]:
+        if self.is_easy:
+            return [self.result, self.result[::-1]]
+        else:
+            return [self.result]
+
+    def payout_ratio(self, table: Table) -> int:
+        payout_type = "easy" if self.is_easy else "hard"
+        return table.settings["hop_payouts"][payout_type]
+
+    @property
+    def _placed_key(self) -> typing.Hashable:
+        return type(self), self.result
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.result}, amount={self.amount})"
```

Comment:
The main hop bet needs customized implementation because it looks specifically at the dice result (not the "number" like many other bets)